### PR TITLE
ASoC: rt715: Remove unused control and modify function prototype rega…

### PR DESCRIPTION
**Pull request by jack-cy-yu**
…rding to warning message.

Remove unused control and modify function prototype regarding to warning message.

### CODE REVIEW

File: sound/soc/codecs/rt715.c [hunk 1]
```diff
@@ -371,7 +370,7 @@ static const char * const adc_22_23_mux_text[] = {
  * 1 will be connected to the same dmic source, therefore we skip index 1 to
  * avoid misunderstanding on usage of dapm routing.
  */
-static const unsigned int rt715_adc_24_25_values[] = {
+static int rt715_adc_24_25_values[] = {
```
Comment (lyakh) [inline sound/soc/codecs/rt715.c:373]: is this change really needed? Strangely, I see both examples in some codecs:
```
static const unsigned int asp_tx_src_values[] = {
	0x00, 0x08, 0x18, 0x19, 0x20, 0x28, 0x29
};

static SOC_VALUE_ENUM_SINGLE_DECL(asp_tx1_src_enum, CS35L36_ASP_TX1_SEL, 0,
				  CS35L36_APS_TX_SEL_MASK, asp_tx_src_text,
				  asp_tx_src_values);
```
and also
```
static int hp_in_mux_map_value[] = {
	0x0, 0x1, 0x2, 0x3,
};

static SOC_VALUE_ENUM_SINGLE_DECL(hpl_in_mux_map_enum,
				  MT6351_AUDDEC_ANA_CON0,
				  RG_AUDHPLMUXINPUTSEL_VAUDP32_SFT,
				  RG_AUDHPLMUXINPUTSEL_VAUDP32_MASK,
				  hp_in_mux_map,
				  hp_in_mux_map_value);
```
but the actual definition of that macro assigns the last argument to
```
struct soc_enum {
	...
	const unsigned int *values;
```
so `const unsigned` seems correct to me.

Comment (plbossart) [reply sound/soc/codecs/rt715.c:373]: Ah, this is a nice one. The warnings comes from the kernel-doc parts, not from the C compilation.

The following diff works from me:

````diff
diff --git a/sound/soc/codecs/rt715.c b/sound/soc/codecs/rt715.c
index 5c6f05b8d8ab..4c746938a062 100644
--- a/sound/soc/codecs/rt715.c
+++ b/sound/soc/codecs/rt715.c
@@ -203,7 +203,6 @@ static int rt715_set_amp_gain_get(struct snd_kcontrol *kcontrol,
        return 0;
 }
 
-static const DECLARE_TLV_DB_SCALE(out_vol_tlv, -6525, 75, 0);
 static const DECLARE_TLV_DB_SCALE(in_vol_tlv, -1725, 75, 0);
 static const DECLARE_TLV_DB_SCALE(mic_vol_tlv, 0, 1000, 0);
 
@@ -366,7 +365,7 @@ static const char * const adc_22_23_mux_text[] = {
        "DMIC4",
 };
 
-/**
+/*
  * Due to mux design for nid 24 (MUX_IN3)/25 (MUX_IN4), connection index 0 and
  * 1 will be connected to the same dmic source, therefore we skip index 1 to
  * avoid misunderstanding on usage of dapm routing.
@@ -404,6 +403,7 @@ static SOC_ENUM_SINGLE_DECL(
 static SOC_VALUE_ENUM_SINGLE_DECL(rt715_adc24_enum,
        RT715_MUX_IN3, 0, 0xf,
        adc_24_mux_text, rt715_adc_24_25_values);
+
 static SOC_VALUE_ENUM_SINGLE_DECL(rt715_adc25_enum,
        RT715_MUX_IN4, 0, 0xf,
        adc_25_mux_text, rt715_adc_24_25_values);
````

will send patch upstream
